ext/url/00_url.js

```diff
@@ -327,8 +330,14 @@
           context: "Argument 2",
         });
       }
+      const components = hrefComponentsMap.get(url);
+      if (base === undefined && components) {
+        this[_url] = { ...components };
+      } else {
+        this[_url] = opUrlParse(url, base);
+        hrefComponentsMap.set(this[_url].href, this[_url]);
```

Comment:
I'm in favour of a LRU cache. btw won't this explode without a certain limit?

Reply:
it would. unsure what we should do about that

Reply:
Maybe?
```js
if (hrefComponentsMap.size > 20) hrefComponentsMap.clear();
hrefComponentsMap.set(this[_url].href, this[_url]);
```

Reply:
hm, ideally it wouldnt be clear but instead deletion of the oldest

Reply:
@littledivy added a clear; dont think there is a performant way to delete only the first entry in a map

Reply:
> @littledivy added a clear; dont think there is a performant way to delete only the first entry in a map

The first entry in the map should be `hrefComponents.keys().next().value`.

```javascript
if (hrefComponents.size > 30) {
 const keyOfFirstEntry = hrefComponents.keys().next().value;
 hrefComponents.delete(keyOfFirstEntry);
}
```

Reply:
> > @littledivy added a clear; dont think there is a performant way to delete only the first entry in a map
> 
> The first entry in the map should be `hrefComponents.keys().next().value`.
> 
> ```js
> if (hrefComponents.size > 30) {
>  const keyOfOldestEntry = hrefComponents.keys().next().value;
>  hrefComponents.delete(keyOfOldestEntry);
> }
> ```

However, that will not be the most ancient one. It would be an option to do order modifying with delete + set calls on existing item access, in which case the first value will correctly be the most ancient

Reply:
> However, that will not be the most ancient one. It would be an option to do order modifying with delete + set calls on existing item access, in which case the first value will correctly be the most ancient 

Why is this the case?
Is this,
```js
hrefComponentsMap.set(this.#serialization, componentsBuf);
```
expected to be encountering duplicate keys? In that case, yes, I understand and agree that my suggestion would not get the oldest key.

Reply:
Yeah, most likely duplicates will exist.